### ISSUE TITLE
Don't automatically start on startup / have an option to disable this

### ISSUE THREAD

**Issue by gregsadetsky** [ollama/ollama#162]
EDIT: [there's a PR for it now](https://github.com/ollama/ollama/pull/7097)!

---

Thanks for making this!

I noticed that on macOS (I suppose it's the same on Windows), the app sets itself to open at login. This is done here:

https://github.com/jmorganca/ollama/blob/91cd54016c47b71223e8263c44250766874e05cf/app/src/index.ts#L175,L180

1) I'm not sure whether that should be the case by default (up for discussion)
2) I don't remember seeing a warning about this when installing the app (this might be debatable as well)
3) it would be great to have an option within the app to disable this

Cheers

**Comment by mchiang0610** [comment]
Sorry about that. We will definitely have a setting for disabling automatic start at login. 

The quickest workaround right now, before we get to a fix, is to disable is via macOS system preferences:

We do respect its settings. 

<img width="709" alt="Image 2023-07-21 at 9 23 19 AM" src="https://github.com/jmorganca/ollama/assets/3325447/6dfe458a-b3a3-4b24-81ef-b05742d7fb15">

**Comment by csterritt** [comment]
Note that this should integrate with [bug 283](https://github.com/jmorganca/ollama/issues/283) wherein it prompts to install the command line tools, even if they're installed.

The problem is that if you quit out of ollama without doing that install, the next time it starts up it sets itself to open at startup again.

**Comment by panckreous** [comment]
While the os-level setting is definitely respected once we undo it at the os level, I'm hoping to see this changed to an opt-in setting addressed through the application. 

In my opinion, it's ok to ask me if I'd like the application to launch on system startup during the install process (checkbox should be off by default), then otherwise I expect to be able to find the setting in the application's preferences. 

I find this to be common practice amongst the higher-quality macos applications I use, especially those with menubar-only presences like ollama. 

In terms of examples, I find that most of the prodigious Sindre Sorhus' newer menubar apps to be good examples ([Velja](https://sindresorhus.com/velja) and [Folder Peek](https://sindresorhus.com/folder-peek), for example, which I assume use his [LaunchAtLogin](https://github.com/sindresorhus/LaunchAtLogin) / [LaunchAtLogin-Modern](https://github.com/sindresorhus/LaunchAtLogin-Modern) libraries to implement) and find almost every single aspect of [CleanShot X](https://cleanshot.com/) to be the gold standard


Most importantly: thank all of you for ollama and all the work you continue to put into it

**Comment by TehPers** [comment]
Pretty sure you can change your startup programs in Windows by opening up Task manager. Make sure to open up the advanced view, and there should be a tab for startup apps. (This should be true of old and new Task Manager)

I'll admit I run this through Docker though and don't have any issues with it opening on startup on Windows as a result, so I haven't been able to verify it's there.

**Comment by RockyHandsome8** [comment]
It doesn't show on my PC, So is there any other way?

**Comment by hichemfantar** [comment]
@mchiang0610 any plans to implement this soon?

**Comment by data-miner00** [comment]
I'm on Linux and ollama is also set to start on boot..

**Comment by johannesrave** [comment]
i'm on ubuntu and i'd also like to turn automatic start off

**Comment by jmorganca** [comment]
For Linux you can try `sudo systemctl disable ollama`. Let me know if this doesn't work.

**Comment by bvwidt** [comment]
@RockyHandsome8 and other Windows users: Ollama doesn't show up in the Task Manager's Startup apps.

To let Ollama not start at Windows startup...
1. go into the startup folder (press `Windows` + `R` -> type `shell:startup` -> click `OK` / press `enter`).
2. delete the shortcut to Ollama.
3. Be happy.

**Comment by nobleinteractive** [comment]
It's ridiculous that I have to manually delete the shortcut in the startup directory after every Ollama update. Is there no way for Ollama to respect this preference on Windows?

**Comment by hichemfantar** [comment]
my PR above should fix the issue on macos but i don't understand why the electron app isn't also being used on windows and linux. the entire point of electron is to have one app that targets all platforms.

**Comment by hichemfantar** [comment]
I'd appreciate if y'all would upvote and comment on this PR that adds a way toggle auto start behavior to speed up the merge process https://github.com/ollama/ollama/pull/7097

**Comment by hopperelec** [comment]
+1 for an option to make it not add itself to start-up on Windows after every update. Everywhere I look just suggests disabling it on start-up using task manager or deleting the shortcut, but these are just temporary solutions. I understand it adding itself to start-up when it's first installed, but surely there should be a special case for updates to an existing installation to prevent it from trying to add itself again?

**Comment by UdoD5** [comment]
+1 Please add an option in preferences to disable or enable autostart (in any OS).

**Comment by sanathu-simpplr** [comment]
+1 Please add an option in preferences to disable or enable autostart (in any OS).

**Comment by Alrecenk** [comment]
+1 Very annoyed that this opened on startup without asking or any indication that it would do so.

**Comment by Hullow** [comment]
On OS X, the app still re-adds itself as a log-in item each time we open it (same issue with running it from a terminal). Annoying bug

**Comment by siviae** [comment]
This is very annoying!

**Comment by benediktask** [comment]
still happening

**Comment by hichemfantar** [comment]
@gregsadetsky can you link my PR directly in the post for better visibility?

**Comment by hichemfantar** [comment]
@gregsadetsky i already did but i feel it would get more attention if you include it in the post itself.
https://github.com/ollama/ollama/pull/7097

**Comment by softwaresinc** [comment]
You have lost my trust before starting. The current setup acts like a trojan.

**Comment by slyyyle** [comment]
This really needs to be reworked.  At first I thought I was just new to this sort of programming, but after moving around for awhile it truthfully is just challenging for a user.  After spending a few weeks just living in Ollama - things need major rework - especially given applications rest on top of it.  Everyone is just throwing their hands up acting like OS management is everyone else's business when in reality, the burden of this management is easily and often taken by developers.  Otherwise PCs would be a mess and no one would use them.

1. Recommending making ollama a system service is mind boggling.  I have to stop the service and restart it because I simply can't access it via API unless I deliberately set host in command line.  Ollama as an automatic service is a production decision that shouldn't be forced on the user.  When I OLLAMA_HOST=Blah ollama serve, it locks my terminal input, so no, running as a service does not seem to be equivalent.

2. Literally 5 minutes of chatGPT generation could make something better than "Ollama is running" which just wastes your port completely.

I understand these aren't made for production environments.  No one is holding developers responsible for this.  It doesn't mean you have to put so much of the burden on the dev that they have to know enough to be able to rewrite the package to use it.  I simply don't understand this culture.

**Comment by evemilano** [comment]
> You have lost my trust before starting. The current setup acts like a trojan. Moved to trash.

I agree 100%, the overall quality of the installation process is close to zero.

**Comment by LingkKang** [comment]
Temporary workaround for Win (manually):

Go to `C:\Users\<UserName>\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\Startup`, and delete the Ollama shortcut.

**Comment by iwinux** [comment]
For macOS users, `brew install --formula ollama` is a way better install method than official app. The Electron app is wasting precious RAM which should be used for loading models instead.

**Comment by hichemfantar** [comment]
The year is 2067. All world governments have fallen. Ollama-branded killer drones patrol the skies, searching for members of the run-on-startup resistance. At last, the final safehouse is located.

As the rebels are dragged into the prison transport by shock troops, one cries out "It wasn't supposed to be like this! You promised we'd be able to disable automatic run-on-startup!"

The eyes on the commander's helmet light up as he replies:

> "Sorry about that. We will definitely have a setting for disabling automatic start at login.
The quickest workaround right now, before we get to a fix, is to disable via macOS system preferences:
We do respect its settings."

Then he slams the door.

**Comment by ReddotCleaner** [comment]
is it so hard to add this feat?

**Comment by JELSTUDIO** [comment]
> Temporary workaround for Win (manually):
> 
> Go to `C:\Users\<UserName>\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\Startup`, and delete the Ollama shortcut.

Sadly Ollama puts in a new short-cut in this folder every time it updates itself.

Isn't there an actual setting that permanently will keep Ollama from auto-starting?

**Comment by clicktodev** [comment]
you should find it in the task manager Startup tab

**Comment by gregsadetsky** [comment]
Hey @dhiltgen - I don't see how the updated documentation solves this issue.

Ollama still installs itself at login, still doesn't let the user know it does this, and still doesn't have an option to disable this.

A few months ago @mchiang0610 was [saying](https://github.com/ollama/ollama/issues/9078#issuecomment-2658455452):

> We are fixing this in a future mac app update.
> 
> Will refer to this issue: [#162](https://github.com/ollama/ollama/issues/162)

(referring to this current issue that was closed)

---

Screenshots below from the latest version:

<img width="300" alt="Image" src="https://github.com/user-attachments/assets/2b233a08-6b0c-45f3-819d-317f13f93a19" />

<img width="300" alt="Image" src="https://github.com/user-attachments/assets/a0722695-0bd8-4da3-8967-ecc6792dc8a0" />

<img width="400" alt="Image" src="https://github.com/user-attachments/assets/8d812f42-c03d-404d-9c12-ccf3818c9b64" />

**Comment by hopperelec** [comment]
For what it's worth, the issue does appear to be resolved for Windows. I've just updated to the latest version, and it did not add itself back to start-up! I'm not sure when this got resolved, because I can't seem to find any recently merged PRs mentioning it, but it can't have been long ago because I was only one minor version behind. I can't speak for MacOS unfortunately. And ideally it should be a UI option rather than having to delete the start menu shortcut.

**Comment by clicktodev** [comment]
so does the new version still auto run on startup or was this behavior removed completely?

**Comment by gregsadetsky** [comment]
@clicktodev it still auto runs on startup on macos

you can see it in the very latest code here - https://github.com/ollama/ollama/blob/main/macapp/src/index.ts#L267

**Comment by clicktodev** [comment]
that's the code for the old electron app, the new app is swift based

**Comment by clicktodev** [comment]
Is the new app closed source? because I can't find the new native app's folder

**Comment by swyxio** [comment]
its been 2 years and this option has not been added lmao

https://github.com/ollama/ollama/pull/11334/files

fix is here, for others looking.

## How can I stop Ollama from starting when I login to my computer

Ollama for Windows and macOS register as a login item during installation.  You can disable this if you prefer not to have Ollama automatically start.  Ollama will respect this setting across upgrades, unless you uninstall the application.

**Windows**
- Remove `%APPDATA%\Microsoft\Windows\Start Menu\Programs\Startup\Ollama.lnk`

**MacOS Monterey (v12)**
- Open `Settings` -> `Users & Groups` -> `Login Items` and find the `Ollama` entry, then click the `-` (minus) to remove

**MacOS Ventura (v13) and later**
- Open `Settings` and search for "Login Items", find the `Ollama` entry under "Allow in the Background`, then click the slider to disable.

**Comment by loky-lp** [comment]
I report that ollama is not present in the login items, yet it started itself when i booted my mac
<img width="393" height="357" alt="Image" src="https://github.com/user-attachments/assets/68b69588-99e2-4291-8481-0ef9436ad4d2" />

**Comment by dhiltgen** [comment]
@loky-lp you might be hitting a bug where putting the app in the Trash doesn't sever the start-at-login behavior.  We're tracking that in #1925 and working on a fix.  Until we get it fixed, emptying your Trash should resolve it.

**Comment by clicktodev** [comment]
@dhiltgen any plans to open source the new desktop app?

**Comment by hutcho66** [comment]
> [@loky-lp](https://github.com/loky-lp) you might be hitting a bug where putting the app in the Trash doesn't sever the start-at-login behavior. We're tracking that in [#1925](https://github.com/ollama/ollama/issues/1925) and working on a fix. Until we get it fixed, emptying your Trash should resolve it.

No, Ollama is still in my Applications folder and yet also doesn't appear in login items. 

Blocking it from all background activity works, but this still seems like a bug if it's attempting to start on login but isn't even in the login apps.

<img width="636" height="97" alt="Image" src="https://github.com/user-attachments/assets/42f06b8b-57f6-4dc7-bb13-ac30c26ff1e3" />

<img width="469" height="262" alt="Image" src="https://github.com/user-attachments/assets/d9960030-7ef9-434b-9be4-2b10044bc7b6" />

**Comment by dhiltgen** [comment]
@hutcho66 what MacOS version are you running?

**Comment by hutcho66** [comment]
15.6 (24G84), also observed on 15.5 before I upgraded.

**Comment by dhiltgen** [comment]
@hutcho66 if you scroll down to `Allow in the Background` is there an Ollama entry there?

**Comment by nico-arianto** [comment]
[Info]

I'm in macOS version `15.6 (24G84)`, I cannot find the `Ollama` in `Open at Login`, but I can find the `Ollama` in `Allow in the Background`.

Q: Do we need to disabled `Ollama` in the `Allow in the Background` to disabled auto start at login?

**Comment by hutcho66** [comment]
> [@hutcho66](https://github.com/hutcho66) if you scroll down to `Allow in the Background` is there an Ollama entry there?

Yes, I mentioned that in my original comment ("Blocking it from all background activity works").

This would seem to be non-obvious behaviour though, which is why I left the comment. Does blocking it from background activity not affect behaviour when I do actually manually open the app?

**Comment by loky-lp** [comment]
@dhiltgen I confirm that emptying the trash doesn't change the situation, like @hutcho66 did.
If it helps I installed Ollama through Homebrew on an M2 Pro running macOS 15.5 (24F74).

**Comment by the-vampiire** [comment]
> Does blocking it from background activity not affect behaviour when I do actually manually open the app?

this is what id like to know. my intuition says turning that off means "this wont work unless its in the foreground" when started manually. but maybe its a hack to disable auto-launching that isnt respecting launch settings? fwiw i dont see ollama in launchagents or launchdaemons (`{~,}/Library/{LaunchAgents,LaunchDaemons}` (local or sys) so now I’m generally curious how this is happening

**Comment by o6uoq** [comment]
I had this same issue, this is what removed it from automatically starting on reboot/login as it did NOT appear in login items:

- Removed the app from *Allow in the background* 
- Ran `brew reinstall --cask ollama-app --force`
- Rebooted

**Comment by clicktodev** [comment]
A simple toggle in the app ui would be nice.
If the team doesn't have the bandwidth then you can open source it and I'll be more than happy to open a pr.

**Comment by jamesford42** [comment]
Two years later ..

This also doesn't exist for windows btw and having some ai thing running constantly all the time on your computer (even if its literally no performance when not in use) is also the sort of thing no one is going to trust you on or want, that isn't using it every day.

Having to google this and find it as a startup application (windows) is fairly bizarre. I actually briefly forgot those even existed still. 

Suggestion:

*The absolute minimum effort thing you could easily do.* 
In the in-app settings list, add an item that literally just explains how to do it manually.

*Even better, if you have more than 2 seconds to work on this ever:*
Add a context menu item for the system tray icon like "change startup behavior" that literally just links to a github page with instructions.

*Alternately you know ... Just stop actually doing this*. 
If someone wants the app to autostart, let them look up how to do it themselves.

**Comment by faybar** [comment]
A lot of comments about respecting the macOS Login Item settings, but actually not respected.

Auto start should be opt-in and an Ollama setting to enable/disable easily. What's the motivation to autostart in a sneaky way? This is a dark pattern. Please fix. Thanks.

**Comment by diegochine** [comment]
This is pretty bad from Ollama's team, I agree with @faybar in that this is a dark pattern. The community has been asking (for years now) for a simple toggle in settings, not huge refactorings or new features, and someone even prepared a full PR for it that was never merged. Instead, two years later we end up with a closed source app. This is an unnecessarily sneaky, if not odd, behaviour. 

I'm uninstalling Ollama and will discourage people from using it. For people looking for alternatives, there are several worth checking out, such as vLLM, TGI and Aphrodite.

**Comment by javifmz** [comment]
In my case, it turned out that the **Raycast** app was triggering Ollama on launch. Disabling Raycast AI stopped it from auto-starting.

**Comment by twentworth** [comment]
I would love to be able to disable this.  I've disabled it from the system preferences many times, but Ollama keeps reactivating it.

**Comment by tonny1983** [comment]
[Here](https://www.reddit.com/r/ollama/comments/1lx45fg/ollama_auto_start_despite_removed_from_open_at/) described a solution which works on my envrionment.

**Comment by davidus05** [comment]
Crazy to see that this is still a thing/an issue.

Removing the toggle for Ollama at "App Background Activity" in macOS Settings > "General" > "Login Items" fixed that issue.

**Comment by rmx90210** [comment]
Asking in earnest: is there a reason this still hasn't been implemented three years later? People are frustrated and [showing each other](https://www.reddit.com/r/ollama/comments/1lujfpb/ollama_auto_start_despite_removed_from_open_at/) how to edit the `.plist`. A toggle exposed in the UI would be a very welcome change 🙏

**Comment by Ninjalope** [comment]
+1 This simple change is necessary for what's supposed to be an open platform.

**Comment by colinvn** [comment]
> [Here](https://www.reddit.com/r/ollama/comments/1lx45fg/ollama_auto_start_despite_removed_from_open_at/) described a solution which works on my envrionment.

The method of modifying the plist file works for me - but only until Ollama updates to a never version, which seems to overwrite the file (because it lives inside the application's content's folder, not in the usual ~/Library folder)

**Comment by rick-github** [comment]
https://github.com/ollama/ollama/blob/main/docs/faq.mdx#how-can-i-stop-ollama-from-starting-when-i-login-to-my-computer

**Comment by akoblentz** [comment]
> https://github.com/ollama/ollama/blob/main/docs/faq.mdx#how-can-i-stop-ollama-from-starting-when-i-login-to-my-computer

I had already done this and still found it starting on boot.
